app/controllers/concerns/idv_step_concern.rb

```diff
@@ -134,4 +134,8 @@ def url_for_latest_step
     step_info = flow_policy.info_for_latest_step
     url_for(controller: step_info.controller, action: step_info.action)
   end
+
+  def undo_steps_from!(step:)
```

Comment:
Now that we have a helper method here, we can change it to `undo_steps_from_controller` (better name sought!) and pass in `self.class` the way we do for `confirm_step_allowed` earlier in this file, and then use flow_policy.controller_to_key to implement it.

Reply:
Nice! This makes it much cleaner.